src/app/courses/view-courses/courses-view-detail.component.ts

```diff
@@ -21,10 +22,24 @@ export class CoursesViewDetailComponent {
   template: `
     <h3>{{data.courseDetail.courseTitle}}</h3>
     <planet-courses-detail [courseDetail]="data.courseDetail"></planet-courses-detail>
+    <button class="mat-raised-button primary-color " (click)="routeToCourses(data?.courseDetail._id)">View Course</button>
+    <button class="mat-raised-button mat-warn font-size-1 margin-lr-8" (click)="close()">Close</button>
   `
 })
 export class CoursesViewDetailDialogComponent {
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    private dialogRef: MatDialogRef<CoursesViewDetailDialogComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: any) {}
```

Comment:
Please move the `) {}` to a new line.  This makes it clearer to see what dependencies are being added.